File: SoftUni Lessons/Python Development/Python Basics April 2019/Lessons and Problems/11.1 - For Loop More Exercises/02. Hospital.py
total_days = int(input())

doctors = 7
treated_patients = 0
untreated_patients = 0


for day in range(1, total_days + 1):
    if day % 3 == 0:
        if untreated_patients > treated_patients:
            doctors += 1
    patients = int(input())
    if patients <= doctors:
        treated_patients += patients
    elif patients > doctors:
        untreated_patients += patients - doctors
        treated_patients += doctors

print(f"Treated patients: {treated_patients}.")
print(f"Untreated patients: {untreated_patients}.")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/12. УПРАЖНЕНИЕ ЦИКЛИ/For-Loop - More Exercises/02. Hospital.js
function hospital(input) {
    // Mask
    let totalDays = Number(input.shift());

    let docs = 7;  
    let fixed = 0;  // :D
    let broken = 0;  // :D

    for(let day = 1; day <= totalDays; day += 1) {
        if (day % 3 === 0 && broken > fixed) {
            docs += 1;
        }

        patients = Number(input.shift());
        if (patients <= docs) {
            fixed += patients;
        } else {
            broken += patients - docs;
            fixed += docs;
        }
    }

    console.log(`Treated patients: ${fixed}.`);
    console.log(`Untreated patients: ${broken}.`); 
}

hospital([4, 7, 27, 9, 1]);  // Should return:
// Treated patients: 23.
// Untreated patients: 21.

hospital([6, 25, 25, 25, 25, 25, 2]);  // Should return:
// Treated patients: 40.
// Untreated patients: 87.